src/Warpstone/Parsers/BasicParsers.cs

```diff
@@ -354,6 +354,37 @@ public static IParser<IList<T>> OneOrMore<T>(IParser<T> parser)
         public static IParser<T> Or<T>(IParser<T> first, IParser<T> second, params IParser<T>[] parsers)
             => InnerOr(parsers.Prepend(second).Prepend(first));
 
+        /// <summary>
+        /// Creates a parser that fails if the specified parser succeeds.
+        /// </summary>
+        /// <typeparam name="T">The result type of the parser that should fail.</typeparam>
+        /// <param name="not">The parser which, if it succeeds, causes the returned parser to fail.</param>
+        /// <returns>A parser trying the given parser, and failing if it succeeds.</returns>
+        public static IParser<T> Not<T>(IParser<T> not)
+            => new NotParser<T>(not);
+
+        /// <summary>
+        /// Creates a parser that parses the nested parser if the condition parser fails.
+        /// </summary>
+        /// <typeparam name="T1">The result type of condition the parser.</typeparam>
+        /// <typeparam name="T2">The result type of the nested parser.</typeparam>
+        /// <param name="condition">The condition parser. If this parser fails, the nested parser is executed. The entire parser fails otherwise.</param>
+        /// <param name="then">The nested parser. This parser is executed if the condition does not hold.</param>
+        /// <returns>A parser trying the given parser, running the nested parser if the condition fails, or failing if the condition succeeds.</returns>
+        public static IParser<T2> Not<T1, T2>(IParser<T1> condition, IParser<T2> then)
+            => Not(condition, then);
```

Comment:
Not sure if I still like this overload.

Reply:
Today I learned that the test framework does not consider a stack overflow a test failure 👀